bubbletea/tea.go

```diff
@@ -67,32 +120,90 @@ func MiddlewareWithProgramHandler(bth ProgramHandler, cp termenv.Profile) wish.M
 		lipgloss.SetColorProfile(cp)
 		return func(s ssh.Session) {
 			p := bth(s)
-			if p != nil {
-				_, windowChanges, _ := s.Pty()
-				go func() {
-					for {
-						select {
-						case <-s.Context().Done():
-							if p != nil {
-								p.Quit()
-								return
-							}
-						case w := <-windowChanges:
-							if p != nil {
-								p.Send(tea.WindowSizeMsg{Width: w.Width, Height: w.Height})
-							}
-						}
+			programHandler(s, p)
+			sh(s)
+		}
+	}
+}
+
+func programHandler(s ssh.Session, p *tea.Program) {
+	if p != nil {
+		_, windowChanges, _ := s.Pty()
+		go func() {
+			for {
+				select {
+				case <-s.Context().Done():
+					if p != nil {
+						p.Quit()
+						return
+					}
+				case w := <-windowChanges:
+					if p != nil {
+						p.Send(tea.WindowSizeMsg{Width: w.Width, Height: w.Height})
 					}
-				}()
-				if _, err := p.Run(); err != nil {
-					log.Error("app exit with error", "error", err)
 				}
-				// p.Kill() will force kill the program if it's still running,
-				// and restore the terminal to its original state in case of a
-				// tui crash
-				p.Kill()
 			}
-			sh(s)
+		}()
+		if _, err := p.Run(); err != nil {
+			log.Error("app exit with error", "error", err)
 		}
+		// p.Kill() will force kill the program if it's still running,
+		// and restore the terminal to its original state in case of a
+		// tui crash
+		p.Kill()
+	}
+}
+
+// Bridge Wish and Termenv so we can query for a user's terminal capabilities.
+type sshOutput struct {
+	ssh.Session
+	tty *os.File
+}
+
+func (s *sshOutput) Write(p []byte) (int, error) {
+	return s.Session.Write(p)
+}
+
+func (s *sshOutput) Read(p []byte) (int, error) {
+	return s.Session.Read(p)
+}
+
+func (s *sshOutput) Fd() uintptr {
+	return s.tty.Fd()
+}
+
+type sshEnviron struct {
+	environ []string
+}
+
+func (s *sshEnviron) Getenv(key string) string {
+	for _, v := range s.environ {
+		k, v, ok := strings.Cut(v, "=")
+		if ok && k == key {
+			return v
+		}
+	}
+	return ""
+}
+
+func (s *sshEnviron) Environ() []string {
+	return s.environ
+}
+
+func outputFromSession(sess ssh.Session) *termenv.Output {
+	sshPty, _, _ := sess.Pty()
+	_, tty, err := pty.Open()
+	if err != nil {
+		log.Fatal(err)
+	}
+	o := &sshOutput{
+		Session: sess,
+		tty:     tty,
 	}
+	environ := sess.Environ()
+	environ = append(environ, fmt.Sprintf("TERM=%s", sshPty.Term))
+	e := &sshEnviron{environ: environ}
+	// We need to use unsafe mode here because the ssh session is not running
+	// locally and we already know that the session is a TTY.
+	return termenv.NewOutput(o, termenv.WithUnsafe(), termenv.WithEnvironment(e), termenv.WithColorCache(true))
```

Comment:
This is basically a hack and would be as simple as passing the session to termenv with this https://github.com/muesli/termenv/pull/122

Reply:
oh yes that would make it a lot better!